### PR TITLE
sys-boot/grub: update to 2.06

### PR DESCRIPTION
**Pull request by sayanchowdhury**
# sys-boot/grub: update to 2.06

## Testing done

CI Passed (except for few arches) http://jenkins.infra.kinvolk.io:8080/job/container/job/sdk/1009/cldsv/
CI Running
http://jenkins.infra.kinvolk.io:8080/job/container/job/sdk/1013/cldsv/

Reviews from the previous PR https://github.com/flatcar/coreos-overlay/pull/2098

- @krnowak 
https://github.com/flatcar/coreos-overlay/pull/2098#discussion_r951485236 - Fixed
https://github.com/flatcar/coreos-overlay/pull/2098#discussion_r951485634 - Fixed
https://github.com/flatcar/coreos-overlay/pull/2098#discussion_r951504046 - This seems to be required, and this is required to construct IUSE, and MULTIBUILD variant used in the ebuild
https://github.com/flatcar/coreos-overlay/pull/2098#discussion_r951508514 - Fixed this

- @pothos 
https://github.com/flatcar/coreos-overlay/pull/2098#discussion_r951532690 - I've fixed the commits, and have added a README.md file adding a bit of context as well as the list of commits picked.
https://github.com/flatcar/coreos-overlay/pull/2098#issuecomment-1353309790 - Tested and looks good, the downgrade worked successfully but did not affect the grub, which stayed at 2.06

- [x] Changelog entries added in the respective `changelog/` directory (user-facing change, bug fix, security fix, update)
- [ ] Inspected CI output for image differences: `/boot` and `/usr` size, packages, list files for any missing binaries, kernel modules, config files, kernel modules, etc.

### CODE REVIEW

**Comment by sayanchowdhury** [comment]
Another fresh CI builds with the accomodated patches: http://jenkins.infra.kinvolk.io:8080/job/container/job/sdk/1013/

**Review by jepio — APPROVED** [review]
lgtm - lets review the `getenv` thing when we update shim and get secure boot working.

File: sdk_container/src/third_party/coreos-overlay/sys-boot/grub/README.md [hunk 1]
```diff
@@ -0,0 +1,476 @@
+Flatcar uses a patched version of the GRUB, which implements the functionality to
+read the [Flatcar Container Linux partition table](https://www.flatcar.org/docs/latest/reference/developer-guides/sdk-disk-partitions/#partition-table)
+
+## History
+
+CoreOS Container Linux maintained a fork of the [grub](https://github.com/coreos/grub) and then was referenced
+in the coreos-overlay. Any changes were made through [PRs](https://github.com/coreos/grub/pulls?q=is%3Apr+is%3Aclosed) to the grub repository.
+
+When Flatcar was born, a `grub` repo under the flatcar-linux namespace was created
+and referenced in the flatcar's coreos-overlay. Except for a few, now many changes
```
Comment (krnowak) [inline sdk_container/src/third_party/coreos-overlay/sys-boot/grub/README.md:10]:
```suggestion
and referenced in the Flatcar's coreos-overlay. Except for a few, now many changes
```

File: sdk_container/src/third_party/coreos-overlay/sys-boot/grub/README.md [hunk 2]
```diff
@@ -0,0 +1,476 @@
+Flatcar uses a patched version of the GRUB, which implements the functionality to
+read the [Flatcar Container Linux partition table](https://www.flatcar.org/docs/latest/reference/developer-guides/sdk-disk-partitions/#partition-table)
+
+## History
+
+CoreOS Container Linux maintained a fork of the [grub](https://github.com/coreos/grub) and then was referenced
+in the coreos-overlay. Any changes were made through [PRs](https://github.com/coreos/grub/pulls?q=is%3Apr+is%3Aclosed) to the grub repository.
+
+When Flatcar was born, a `grub` repo under the flatcar-linux namespace was created
```
Comment (krnowak) [inline sdk_container/src/third_party/coreos-overlay/sys-boot/grub/README.md:9]: I suppose the namespace is actually the GitHub org, right?

```suggestion
When Flatcar was born, a `grub` repo under the flatcar-linux org was created
```

File: sdk_container/src/third_party/coreos-overlay/sys-boot/grub/README.md [hunk 3]
```diff
@@ -0,0 +1,476 @@
...
+and referenced in the flatcar's coreos-overlay. Except for a few, now many changes
+where brought into the system.
+
+The repo was maintained at 2.02 version. During the 2.06 migration, the philosophy
+to use a separate repo was scraped, and a single patch file was created. The patch
+file migrated only the essentials commits, and dropped all the other commits, which
```
Comment (krnowak) [inline sdk_container/src/third_party/coreos-overlay/sys-boot/grub/README.md:15]: Later it's mentioned that there are two patches, so let's use plural.

```suggestion
files migrated only the essential commits, and dropped all the other commits, which
```

File: sdk_container/src/third_party/coreos-overlay/sys-boot/grub/README.md [hunk 4]
```diff
@@ -0,0 +1,476 @@
...
+were either half-baked, or redundant at the point of migration. The two patch is applied
```
Comment (krnowak) [inline sdk_container/src/third_party/coreos-overlay/sys-boot/grub/README.md:16]:
```suggestion
were either half-baked, or redundant at the point of migration. The two patches are applied
```

File: sdk_container/src/third_party/coreos-overlay/sys-boot/grub/README.md [hunk 5]
```diff
@@ -0,0 +1,476 @@
...
+on top of the grub sources, and emerge is done.
+
+Given below are the list of commits that were referenced to create the two patches
```
Comment (krnowak) [inline sdk_container/src/third_party/coreos-overlay/sys-boot/grub/README.md:19]:
```suggestion
Given below are the list of commits that were referenced to create the two patches.
```

File: sdk_container/src/third_party/coreos-overlay/sys-boot/grub/README.md [hunk 6]
```diff
@@ -0,0 +1,476 @@
...
+
+## Summary of the patches
+
+The patch start with implementing a new implementation of reading the GPT instead
```
Comment (krnowak) [inline sdk_container/src/third_party/coreos-overlay/sys-boot/grub/README.md:23]:
```suggestion
The patch starts with adding a new implementation of reading the GPT instead
```

File: sdk_container/src/third_party/coreos-overlay/sys-boot/grub/README.md [hunk 7]
```diff
@@ -0,0 +1,476 @@
...
+of using the traditional module. It provides essential functionality to interact
+with GPT structures on disk, and checking/validating data integrity & GPT specification.
+
+The commits goes on to add the following modules gptprio, gptrepair, and search
+commands by label and partition.
+
+The `gptprio` command which provides a mechanism to prioritize and select the
+next bootable partition based on the GPT attributed and results in flexible
```
Comment (krnowak) [inline sdk_container/src/third_party/coreos-overlay/sys-boot/grub/README.md:31]:
```suggestion
next bootable partition based on the GPT attributes and results in flexible
```

File: sdk_container/src/third_party/coreos-overlay/sys-boot/grub/README.md [hunk 8]
```diff
@@ -0,0 +1,476 @@
...
+partition booting. The `gptrepair` command implements the repair functions for
+GPT information on a specified device. Few other functions include, searching
```
Comment (krnowak) [inline sdk_container/src/third_party/coreos-overlay/sys-boot/grub/README.md:33]:
```suggestion
GPT information on a specified device. Few other functions include searching
```

File: sdk_container/src/third_party/coreos-overlay/sys-boot/grub/README.md [hunk 9]
```diff
@@ -0,0 +1,476 @@
...
+devices by partition label, and partition UUID.
```
Comment (krnowak) [inline sdk_container/src/third_party/coreos-overlay/sys-boot/grub/README.md:34]:
```suggestion
devices by partition label or partition UUID.
```

File: sdk_container/src/third_party/coreos-overlay/sys-boot/grub/README.md [hunk 10]
```diff
@@ -0,0 +1,476 @@
...
+
+## Commits
+
+Below are the commits that are picked to create the two patch for the grub. One is
```
Comment (krnowak) [inline sdk_container/src/third_party/coreos-overlay/sys-boot/grub/README.md:38]:
```suggestion
Below are the commits that are picked to create the two patches for the grub. One is
```

**Review by krnowak — APPROVED** [review]

**Pull request merged**